elastic/eui
# [EuiInlineEdit] Allow `EuiInlineEdit` to be used as a controlled component

Closes https://github.com/elastic/eui/issues/7084 || Previous PR: https://github.com/elastic/eui/pull/7117

## Summary
- Allows`EuiInlineEditText` and `EuiInlineEditTitle` title to both be used as controlled components by creating three new props (`value`, `onChange`, `onCancel`).
- Updates the props of inline edit & friends to exclusively accept `defaultValue` OR controlled props (`value`, `onChange`, `onCancel`)
- Clean up `EuiInlineEditText` and `EuiInlineEditTitle` by removing unneeded and duplicated props.
- Adds test cases to ensure new logic & controlled props are covered by test cases
<img width="1369" alt="image" src="https://github.com/elastic/eui/assets/40739624/bb9ac10c-1728-43cc-bae1-9577a9efb9dc">

## QA
View the [new controlled component section](https://eui.elastic.co/pr_7157/#/forms/inline-edit#controlling-inline-edit-text-and-title-values) in the Inline Edit docs

- [ ] Confirm you can go from read mode to edit mode by clicking on the read mode button
- [ ] Make changes in edit mode and save -> You should find your new text, not the default `Hello World!`
- [ ] Go back to edit mode to make changes and cancel -> You should find the previous text, not your most recent changes
- There should be **no** visual differences in this example vs. the other inline edit sections

### General checklist
